blog/2022-04-06-SDL-snowpark/2022-04-06-SDL-snowpark.md

```diff
@@ -0,0 +1,260 @@
+
+First we have add SDLBs Snowflake library to the projects pom.xml dependencies section:
+
+      <dependencies>
+        ....
+        <dependency>
+          <groupId>io.smartdatalake</groupId>
+          <artifactId>sdl-snowflake_${scala.minor.version}</artifactId>
+          <version>${project.parent.version}</version>
+        </dependency>
+        ...
+      </dependencies>
+
+Then SDLB version needs to be updated to version 2.3.0-SNAPSHOT at least in the parent section:
+
+      <parent>
+        <groupId>io.smartdatalake</groupId>
+        <artifactId>sdl-parent</artifactId>
+        <version>2.3.0-SNAPSHOT</version>
+      </parent>
+
+## Define Snowflake connection
+
+To define the Snowflake connection in config/application.conf, add connections section with connection "sf-con", and fill in informations according to prerequisits:
+
+      connections {
+        sf-con {
+          type = SnowflakeTableConnection
+          url = "<accountUrl>",
+          warehouse = "COMPUTE_WH",
+          database = "testdb",
+          role = "ACCOUNTADMIN",
+          authMode = {
+            type = BasicAuthMode
+            userVariable = "CLEAR#<username>"
+            passwordVariable = "CLEAR#<pwd>"
+        }
+      }
+
+## Migrate DataObjects
+
+Now we can change the DataObject type to SnowflakeTableDataObject and the new Snowflake connection, adding the definition of the table:
+
+      int-airports {
+        type = SnowflakeTableDataObject
+        connectionId = sf-con
+        table {
+          db = "test"
+          name = "int_airports"
+        }
+      }
+    
+      btl-departures-arrivals-airports {
+        type = SnowflakeTableDataObject
+        connectionId = sf-con
+        table {
+          db = "test"
+          name = "btl_departures_arrivals_airports"
+        }
+      }
+    
+      btl-distances {
+        type = SnowflakeTableDataObject
+        connectionId = sf-con
+        table {
+          db = "test"
+          name = "btl_distances"
+        }
+      }
+
+Note that the attribute `db` of the SnowflakeTableDataObject should be filled with the schema of the Snowflake table and that this is *not* the same as the attribute `database` of SnowflakeTableConnection. 
+
+## Migrating Actions
+
+The new SDLB version introduced some naming changes:
+- The CustomSparkAction can now also process Snowpark-DataFrames and is therefore renamed to CustomDataFrameAction.
+- The ScalaClassDfTransformer was specific for Spark. In the new SDLB version there is a specific scala-class DataFrame transformer for Spark and Snowpark, e.g. ScalaClassSparkDfTransformer and ScalaClassSnowparkDfTransformer. And there is even a ScalaClassGenericDfTransformer to implement transformations using a unified API. In our case we will migrate the transformation to use Snowpark and set the type to ScalaClassSnowparkDfTransformer.
+- See [Architecture](../../docs/architecture) chapter "breaking changes" for other renamings.

```

Comment:
does not exist yet, to be added later

Reply:
I remove it. It should point to github release notes but there the breaking changes for the current Snapshot-Version are not yet released...